kernel/sched.c

```diff
@@ -1419,12 +1420,7 @@ static int32_t z_tick_sleep(k_ticks_t ticks)
 
 	__ASSERT(!arch_is_in_isr(), "");
 
-#ifndef CONFIG_TIMEOUT_64BIT
-	/* LOG subsys does not handle 64-bit values
-	 * https://github.com/zephyrproject-rtos/zephyr/issues/26246
-	 */
-	LOG_DBG("thread %p for %u ticks", _current, ticks);
-#endif
+	LOG_DBG("thread %p for %" K_TICKS_FMT " ticks", _current, ticks);
```

Comment:
Based on the discussion from https://github.com/zephyrproject-rtos/zephyr/issues/46032, we are probably never going to use `PRIxN` macros (or any macros that resemble those for that matter) in the Zephyr codebase.

IMHO, as @keith-packard noted, casting to `(long)` and hard-coding `%ld` is the most reasonable way forward.

```suggestion
	LOG_DBG("thread %p for %ld ticks", _current, (long)ticks);
```

Reply:
Ouch, I did not realize this change could trigger another religious debate. Adjusted to use unsigned long.